### PR TITLE
plot operations

### PR DESCRIPTION
**Pull request by abuts**
This deals with Re#1501 and fixes three bugs identified in plots

1)  --  the one mentioned in ticket  -- incorrect return from plots
2)  -- output from plotting dnd objects when output does not requested (e.g. `plot(d1d())` would print output, though it was not pringing anything in Horace3
3) -- failure in plotting arrays of objects. Some classes from hirachy `[IX_dataset,dnd, sqw]` were plotting them, some were not -- actually the state was unclear so unit tests were written to verify they are plotting.

To solve these issues the code has been refactored to increase code usage and remove all fluff (e.g. unnecessary `feval`) present in the current code. 

There were a lot of functions which contain almost identical code with minimal differences. All identical code blocks have been replaced by calls to appropriate function with different parameters. This increases code reliability and maintanabilty.

If reviewer goes from commit to commit, it becomes obvious that there are identical changes in similar functions

### CODE REVIEW

File: herbert_core/utilities/classes/@IX_dataset_2d/private/plot_2d_nd_oc_.m [hunk 1]
```diff
@@ -0,0 +1,55 @@
+function [fig_,axes_,plot_] = plot_2d_nd_oc_(w,nout,type,opt,varargin)
```
Comment (cmarooney-stfc) [inline herbert_core/utilities/classes/@IX_dataset_2d/private/plot_2d_nd_oc_.m:1]: Additional function name component oc_ is not documented - comments here look cut and paste from the one without oc.

File: herbert_core/utilities/classes/@IX_dataset_2d/paoc.m [hunk 2]
```diff
@@ -1,4 +1,4 @@
-function [fig_handle, axes_handle, plot_handle] = paoc(w)
+function varargout = paoc(w)
```
Comment (cmarooney-stfc) [inline herbert_core/utilities/classes/@IX_dataset_2d/paoc.m:1]: be nice to have an explanation of the function name

File: herbert_core/utilities/classes/@IX_dataset_2d/ds2.m [hunk 3]
```diff
@@ -1,4 +1,4 @@
-function [fig_handle, axes_handle, plot_handle] = ds2(w,varargin)
+function varargout = ds2(w,varargin)
```
Comment (cmarooney-stfc) [inline herbert_core/utilities/classes/@IX_dataset_2d/ds2.m:1]: useful  to say how this differs from ds

File: herbert_core/graphics/plot_interfaces/dnd_plot_interface.m
Comment (cmarooney-stfc) [inline herbert_core/graphics/plot_interfaces/dnd_plot_interface.m:1]: repeat of request to explain the oc suffices on the function names

File: herbert_core/graphics/plot_interfaces/sqw_plot_interface.m
Comment (cmarooney-stfc) [inline herbert_core/graphics/plot_interfaces/sqw_plot_interface.m:1]: repeat of request for explanation of oc suffices on function names

File: _test/common_functions/data_plot_interface_tester.m [hunk 4]
```diff
@@ -8,11 +8,18 @@
 
     properties(Constant)
         dnd_methods = {@plot,@plotover}
+        dnd_overlpot = [false,true];
```
Comment (cmarooney-stfc) [inline _test/common_functions/data_plot_interface_tester.m:11]:
```suggestion
        dnd_overplot = [false,true];
```
also on uses and comments below

File: _test/common_functions/data_plot_interface_tester.m [hunk 5]
```diff
@@ -8,11 +8,18 @@
...
+            @pd,@pe,@pl,@ph,@pm,@pp}
+        d1d_overlpot = [false(1,6),true(1,6)];
+
+        % overlpot only methods
+        d1d_mthods_oveplot = { @pdoc,@peoc,@phoc,@ploc,@pmoc,@ppoc}
         d2d_methods = {@da,@ds,@ds2,...
             @pa,@paoc,@ps,@ps2,@ps2oc,@psoc};
+        overplot_requested = [false,false,false,...
+            false,true,false,false,true,true];
+        d2d_ovelplot = [false(1,3),true(1,6)];
```
Comment (cmarooney-stfc) [inline _test/common_functions/data_plot_interface_tester.m:22]:
```suggestion
        d2d_overplot = [false(1,3),true(1,6)];
```

File: herbert_core/utilities/classes/@IX_dataset_2d/private/plot_2d_nd_oc_.m (hunk 1 above)
Comment (abuts) [reply herbert_core/utilities/classes/@IX_dataset_2d/private/plot_2d_nd_oc_.m:1]: added description

File: herbert_core/utilities/classes/@IX_dataset_2d/ds2.m (hunk 3 above)
Comment (abuts) [reply herbert_core/utilities/classes/@IX_dataset_2d/ds2.m:1]: actually it was there, written at the end of the description

Comment (abuts) [reply herbert_core/utilities/classes/@IX_dataset_2d/ds2.m:1]: but added

**Pull request merged**